example_test.go

```diff
@@ -620,6 +620,11 @@ func ExampleSubOpt() {
 	js.Subscribe("foo", func(msg *nats.Msg) {
 		fmt.Printf("Received a message: %s\n", string(msg.Data))
 	}, nats.StartTime(time.Now().Add(-2*time.Hour)))
+
+	// Start delivering messages with delay based on BackOff array of time durations.
+	js.Subscribe("foo", func(msg *nats.Msg) {
+		fmt.Printf("Received a message: %s\n", string(msg.Data))
+	}, nats.BackOff([]time.Duration{50 * time.Millisecond, 250 * time.Millisecond}))
```

Comment:
This complies, but would actually not "work" because you would need to set the MaxDeliver to at least 2 (or more), but also add `nats.ManualAck()` because otherwise the callback would auto-ack anyway, so it is unlikely that there would be redeliveries...

So instead (or at least in addition to) having this here, I would update, say TestJetStreamSubscribe_AckPolicy test to use the nats.BackOff() new option. In this test (at the end of it), we use backoff but with the AddConsumer directly.